includes/admin/post-types/post-tables/class.llms.admin.post.table.questions.php

```diff
@@ -0,0 +1,430 @@
+ */
+
+if ( ! defined( 'ABSPATH' ) ) { exit; }
+
+class LLMS_Admin_Post_Table_Questions {
+	
+	public $quiz_p_id = 0;
+	public $lesson_p_id = 0;
+
+	/**
+	 * Constructor
+	 * @return  void
+	 * @since    3.9.6
+	 * @version  3.9.6
+	 */
+	public function __construct() {
+	
+		add_filter( 'manage_llms_question_posts_columns', array( $this, 'add_columns' ), 10, 1 );
+		add_action( 'manage_llms_question_posts_custom_column', array( $this, 'manage_columns' ), 10, 2 );
+
+		//add course filter 
+		add_action( 'restrict_manage_posts', array( $this, 'filters' ), 10 );
+
+		//change query
+		add_action( 'pre_get_posts', array( $this, 'query_posts_filter' ), 10,1 );
+
+		//disable default date 
+		add_filter( 'months_dropdown_results', array( $this, 'default_date_filter' ), 10 ,2);
+
+	}
+
+	/**
+	 * Add Custom lesson Columns
+	 * @param   array  $columns  array of default columns
+	 * @return  array
+	 * @since    3.9.6
+	 * @version  3.9.6
+	 */
+	public function add_columns( $columns ) {
+
+		$columns = array(
+			'cb' => '<input type="checkbox" />',
+			'title' => __( 'Lesson Title', 'lifterlms' ),
+			'course' => __( 'Course', 'lifterlms' ),
+			'lesson' => __( 'Lesson', 'lifterlms' ),
+			'quiz' => __( 'Quiz', 'lifterlms' ),
+			//'prereq' => __( 'Prerequisite', 'lifterlms' ),
+			'date' => __( 'Date', 'lifterlms' ),
+		);
+
+		return $columns;
+	}
+
+
+	/**
+	 * Manage content of custom lesson columns
+	 * @param  string $column   column key/name
+	 * @param  int    $post_id  WP Post ID of the lesson for the row
+	 * @return void
+	 * @since   3.9.6
+	 * @version  3.9.6
+	 */
+	public function manage_columns( $column, $post_id ) {
+		//set data starts 
+		$this->quiz_p_id = 0;
+		$this->lesson_p_id  =0;
+		$all_quizes 	=	$this->get_posts('llms_quiz');
+				
+				foreach($all_quizes  as $q_id) {
+					
+					//get questions of quiz 
+					$q = array(); 
+					$q = get_post_meta( $q_id, '_llms_questions', true );
+					$r = wp_list_pluck( $q, 'id' );
+				
+					//check if current question exists this quiz quiz
+					if(in_array($post_id,$r)){
+						$edit_link = get_edit_post_link( $q_id );
+						//set quiz id 
+						$this->quiz_p_id = $q_id;
+												
+					}
+					
+				}
+				
+				$all_less 	=	$this->get_posts('lesson');
+				foreach($all_less  as $lesson_id) {
+					
+					$quiz_id = absint( get_post_meta( $lesson_id, '_llms_assigned_quiz', true ) );
+					if($this->quiz_p_id){
+						if($quiz_id == $this->quiz_p_id){
+							
+							$edit_link = get_edit_post_link( $lesson_id );
+							//set lesson id 
+							$this->lesson_p_id = $lesson_id;
+							}
+						}
+					}
+				
+		
+		//set data ends 
+		
+
+		switch ( $column ) {
+			case 'course' :
+
+				
+					$parent_id = absint( get_post_meta( $this->lesson_p_id, '_llms_parent_course', true ) );
+					$edit_link = get_edit_post_link( $parent_id );
+
+						if ( ! empty( $parent_id ) ) {
+							printf( __( '<a href="%1$s">%2$s</a>' ), $edit_link, get_the_title( $parent_id ) );

```

Comment:
__() is unnecessary here since only variables are being passed into the l10n func